.github/scripts/createHelpRedirects.sh

```diff
@@ -0,0 +1,121 @@
+    # subdomains.
+    if ! [[ $SOURCE_URL =~ ^https://community\.expensify\.com ]]; then
+        error "Found source URL that is not a community URL: $SOURCE_URL"
+        exit 1
+    fi
+
+    if ! [[ $DEST_URL =~ ^https://help\.expensify\.com ]]; then
+        error "Found destination URL that is not a help URL: $DEST_URL"
+        exit 1
+    fi
+
+    info "Source: $SOURCE_URL and destination: $DEST_URL appear to be formatted correctly."
+
+    ITEMS_TO_ADD+=("$line")
+
+# This line skips the first line in the csv because the first line is a header row.
+done <<< "$(tail +2 $REDIRECTS_FILE)"
+
+# Sanity check that we should actually be running this and we aren't about to delete
+# every single redirect.
+if [[ "${#ITEMS_TO_ADD[@]}" -lt 1 ]]; then
+    error "No items found to add, why are we running?"
+    exit 1
+fi
+
+# This block builds a single JSON object with all of our updates so we can
+# reduce the number of API calls we make. You cannot add any logging or anything
+# that prints to std out to this block or it will break. We capture all of the std out
+# from this loop and pass it to jq to build the json object. Any non-json will break the
+# jq call at the end.
+PUT_JSON=$(for new in "${ITEMS_TO_ADD[@]}"; do
+    read -r -a LINE_PARTS < <(echo "$new" | tr ',' ' ')
+    SOURCE_URL=${LINE_PARTS[0]}
+    DEST_URL=${LINE_PARTS[1]}
+    jq -n --arg source "$SOURCE_URL" --arg dest "$DEST_URL" '{"redirect": {source_url: $source, target_url: $dest}}'
+done | jq -n '. |= [inputs]')
+
+info "Adding redirects for $PUT_JSON"
+
+# We use PUT here instead of POST so that we replace the entire list in place. This has many benefits:
+# 1. We don't have to check if items are already in the list, allowing this script to run faster
+# 2. We can support deleting redirects this way by simply removing them from the list
+# 3. We can support updating redirects this way, in the case of typos or moved destinations.
+#
+# Additionally this API call is async, so after we finish it, we must poll to wait for it to finish to
+# to know that it was actually completed.
+PUT_RESULT=$(curl -s --request PUT --url "https://api.cloudflare.com/client/v4/accounts/$ZONE_ID/rules/lists/$LIST_ID/items" \
+    --header 'Content-Type: application/json' \
+    --header "Authorization: Bearer $CLOUDFLARE_LIST_TOKEN" \
+    --data "$PUT_JSON")
+
+checkCloudflareResult "$PUT_RESULT"
+OPERATION_ID=$(echo "$PUT_RESULT" | jq -r .result.operation_id)
+
+DONE=false
+
+# Poll for completition
+while [[ $DONE == false ]]; do
+    CHECK_RESULT=$(curl -s --request GET --url "https://api.cloudflare.com/client/v4/accounts/$ZONE_ID/rules/lists/bulk_operations/$OPERATION_ID" \
+        --header 'Content-Type: application/json' \
+        --header "Authorization: Bearer $CLOUDFLARE_LIST_TOKEN")
+    checkCloudflareResult "$CHECK_RESULT"
+
+    STATUS=$(echo "$CHECK_RESULT" | jq -r .result.status)
+
+    # Exit on completed or failed, other options are pending or running, in both cases
```

Comment:
We don't need to add a max retry/time limit to this, right?

Reply:
I don't think so? We could but... i'm inclined to just let it run like this until it becomes a problem. Unless the Cloudflare API just never ran the job and left it queued forever, i think we will eventually exit here.